### ISSUE TITLE
Results problem on aircraft dataset

### ISSUE THREAD

**Issue by successhaha** [sungnyun/openssl-simcore#5]
Hi, I have read your paper repeatedly and that is a good job. At present, I am trying to replicate your results. However, I have encountered some problems in the process of replicating your results, which are as follows:
Following your experimental settings, I first conducted ssl training for Aircraft(train) by using simclr method. Then, test its performance on X(Aircraft test) using run_selfup.sh. However, the test results on X(Aircraft) are only [20.55, 28.32, 90.25](Acc@1, Acc@5, Train Acc).
There is a clear difference between this result and the paper.
Could you possibly give some suggestions as to the cause of this gap? 
The specific training parameters I use are as follows:
```
TAG="aircraft_vanilla"
DATA="aircraft"

CUDA_VISIBLE_DEVICES=0,1,2,3 python train_selfsup.py --tag $TAG \
    --no_sampling \
    --model resnet50 \
    --batch_size 512 \
    --precision \
    --dataset $DATA \
    --data_folder ./data \
    --method simclr \
    --epochs 5000 \
    --cosine \
    --learning_rate 1e-1 \
    --weight_decay 1e-4
 ```

**Comment by raymin0223** [comment]
Hi, wei xiao.
Thanks for your interest in our paper.

Strangely, your script looks exactly the same as ours.
We are now comparing our current codes with codes before refactoring.

For clarity, I would like to check if your Aircraft(train) is the union of train and val of the original Aircraft dataset.
And can you give us run_sup.sh as well?

Thanks,
Sangmin

**Comment by successhaha** [comment]
Thank you very much for your reply. I'll check the generative part of the Aircraft dataset right away. Also, this is the content of the run_sup.sh I used as follows:
‘’‘
TAG="aircraft_vanilla"
DATA="aircraft"

python train_selfsup.py --tag $TAG \
    --no_sampling \
    --model resnet50 \
    --batch_size 512 \
    --precision \
    --dataset $DATA \
    --data_folder ./data \
    --method simclr \
    --epochs 5000 \
    --cosine \
    --learning_rate 1e-1 \
    --weight_decay 1e-4
‘’‘

**Comment by raymin0223** [comment]
If you did not merge train and valid dataset of Aircraft, run [`aircraft_image_folder_generator.py`](https://github.com/sungnyun/openssl-simcore/blob/main/data/aircraft_image_folder_generator.py) for the preprocessing.

And I wanted to check your script file for linear evaluation, like [`run_sup.sh`](https://github.com/sungnyun/openssl-simcore/blob/main/run_sup.sh)!
I would like to note that we also used 4 GPUs to run linear evaluation.

**Comment by successhaha** [comment]
hi, I've used aircraft_image_folder_generator.py to generate the train dataset. I used 4 A100 GPUs for the test. The contents of run_sup.sh are as follows:
‘’‘
TAG="aircraft_vanilla"
DATA="aircraft"

python train_sup.py --tag $TAG \
    --dataset $DATA \
    --model resnet50 \
    --data_folder ./data \
    --pretrained \
    --pretrained_ckpt ./save/aircraft_resnet50_pretrain_simclr_aircraft_vanilla/last.pth\
    --method simclr \
    --epochs 100 \
    --learning_rate 10 \
    --weight_decay 0
’‘’
In addition, when I tested the Pets dataset earlier, the results were generally consistent with those in the paper.

**Comment by raymin0223** [comment]
Hi, wei.

Sorry for you to wait, but we couldn't find problems in our training codes yet.
Also, thanks to your Pets experiments, I believe that there is something different in preprocessing Aircraft dataset.

Test accuracy of 20% looks similar to 'manufacturers' experimental results (see Table 7d).
So, can you check if the attribute of Aircraft is set to 'variants' properly?

We will keep finding some bugs in our codes on Aircraft.

**Comment by successhaha** [comment]
Hi,SangminBae.
Thank you very much for your reply. I am very interested in your work, and I will do more interesting experiments. I'm going to double-check my experimental settings. I also think that this result may be due to my experimental setup, which obviously overfits the training set.

**Comment by raymin0223** [comment]
Sorry for this inconvenience.
We will also run the experiments again, and let you know after we get the results or find some bugs!

**Comment by raymin0223** [comment]
Hi, @successhaha.

I've run Aircraft experiments with your script files from the scratch.
And, I got the results of `best accuracy: [47.59, 78.43, 94.59] (Acc@1, Acc@5, Train Acc)`.

I assumed that the difference is due to the dataset settings.
Thereby, I uploaded Aircraft preprocessed dataset and pretrained checkpoint in this [link](https://drive.google.com/file/d/15d2XpYhhGJuDKw7CqeKqf17EfBEMrHym/view?usp=share_link).

Can you download them and check the results?
Please let me know if you can reproduce the right results!

**Comment by successhaha** [comment]
Thank you very much for your reply. I have repeated the same result [47.17, 78.52, 94.78] (Acc@1, Acc@5, Train Acc).
I have also found the problem of experimental Settings. For Aircraft, it is necessary to distinguish the Variant, Family and Manufacturer. Thanks again for your reply, I will close this issue.

**Comment by raymin0223** [comment]
We should have clarified the preprocessing codes about 'multi-attribute' settings. 🥲
We will update later, thanks for reporting issues!!

**Comment by prajwalsingh** [comment]
@raymin0223 @successhaha  Thanks a lot. I came here to ask the same question and got the answers.

**Comment by raymin0223** [comment]
Hi @prajwalsingh, sorry for the late reply.

The reason was preprocessing of Aircraft dataset, regarding to multi-attribute.
The default attribute setting of Aircraft is 'variants', NOT 'manufacturers' like in [aircraft_image_folder_generator.py](https://github.com/sungnyun/openssl-simcore/blob/main/data/aircraft_image_folder_generator.py).

Can you try with 'variants' attribute?
Also, you can find both variants-preprocessed data and pretrained checkpoint in this [link](https://drive.google.com/file/d/15d2XpYhhGJuDKw7CqeKqf17EfBEMrHym/view?usp=share_link).

**Comment by prajwalsingh** [comment]
@raymin0223 Thank you for the reply. Yes I downloaded the pre-processed that you provided in the previous comments. Its working fine now. Yes I checked the data generation code, and variant was missing from there the default setting was manufacturers.

**Comment by raymin0223** [comment]
Sounds good! 
Note that the default attribute for Cars is 'model' (refer to explanation of Table 7d).

I'm closing this issue, but reopen it when you have further questions.

**Comment by successhaha** [comment]
Hi, @raymin0223, thank you very much for your kind answers all the time. ​While doing experiments, I have found that some other problems, 1) the link Cars(Stanford Cars) is broken; 2) the Flowers dataset is missing the file of Imagelabels.mat. Can you provide these data sets? Thank you very much.

**Comment by sungnyun** [comment]
Hi @successhaha, 

1) Looks like the link has been broken by the original source provider. Let me try to find a different way to get it. Since this dataset size is not so big, I might be able to share this by dropbox. Stay tuned just for a few days.
2) It should have. Check this [link](https://www.robots.ox.ac.uk/~vgg/data/flowers/102/imagelabels.mat).

Thank you and ask if you have further questions :)

**Comment by sungnyun** [comment]
@successhaha 
Check this [Dropbox link](https://www.dropbox.com/s/t8p2cjok9i5lxyg/cars.tar.gz?dl=0) for the cars dataset.

**Comment by successhaha** [comment]
Hi，@sungnyun
Thank you for your Dropbox link for the Cars dataset, but Dropbox is temporarily unavailable in my area due to some force majeure reasons. Can you provide a Google Drive link? Thank you very much.

**Comment by raymin0223** [comment]
Hi @successhaha,
Can you check this [link](https://drive.google.com/file/d/1Sn1Ysvr-GozYjuxK88n-Opk2J_R9lUJ2/view?usp=share_link)? 
This includes all attribute dataset of Cars, such as 'models' (default), 'brand', and 'type'.

**Comment by successhaha** [comment]
Hi，@raymin0223
Thank you very much, this link can download the Cars dataset.I have no further questions for the time being.
